compliance_tool/cli.py

```diff
@@ -1,57 +1,79 @@
+    #       be the default listing operation.
+    parser.set_defaults(func=list_.execute)
+    list_subparsers = parser.add_subparsers(required=False)
 
     list_subparsers.add_parser("profiles",
                                help="list available profiles")
 
-    parser = list_subparsers.add_parser("controls",
-                                        help="list available controls for a given profile")
-    parser.add_argument("--profile")
+    list_subparsers.add_parser("controls", parents=[profile_parser],
+                               help="list available controls (requirements)"
+                                    " within a given profile")
 
-    parser = list_subparsers.add_parser("rules",
-                                        help="list available rules for a given profile")
-    parser.add_argument("--profile")
+    list_subparsers.add_parser("rules", parents=[profile_parser],
+                               help="list available rules for a given profile")
 
 
-def prepare_parser_scan(subparsers) -> None:
-    parser = subparsers.add_parser("scan",
-                                   help="perform a compliance scan")
+def prepare_parser_scan(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("scan", parents=[global_parser, profile_parser],
+                                        help="perform a compliance scan")
     parser.set_defaults(func=scan.execute)
-    parser.add_argument("--profile")
-    parser.add_argument("--control")
-    parser.add_argument("--rule")
 
-    parser.add_argument("--html")
-    parser.add_argument("--json")
+    parser.add_argument("--control", metavar="CONTROL_ID",
+                        help="use only control(s) (requirement(s)) with given identifier")
+    parser.add_argument("--rule", metavar="RULE_ID",
+                        help="use only rule(s) with given identifier")
 
+    parser.add_argument("--html", metavar="DESTINATION_FILE",
+                        help="write a human-readable HTML report with the results"
+                             " at the given destination")
+    parser.add_argument("--json", metavar="DESTINATION_FILE",
+                        help="write a JSON-formatted report with the results"
+                             " at the given destination")
 
-def prepare_parser_remediate(subparsers) -> None:
-    parser = subparsers.add_parser("remediate",
-                                   help="perform a system remediation")
+
+def prepare_parser_remediate(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("remediate", parents=[global_parser, profile_parser],
+                                        help="perform a system remediation")
     parser.set_defaults(func=remediate.execute)
-    parser.add_argument("--profile")
-    parser.add_argument("--control")
-    parser.add_argument("--rule")
+
+    parser.add_argument("--control", metavar="CONTROL_ID",
```

Comment:
In the specification doc we use the term `requirement` but here is used `control`. We should stick to the specification doc.

Reply:
For now we've decided that control(s) works better. See conversation above.